File: packages/shared-types/src/device.ts
export interface Device {
  id: string;
  workspaceId: string;
  name: string;
  role: 'hub' | 'client' | 'standalone';
  status: 'pending' | 'approved' | 'online' | 'offline' | 'suspended' | 'revoked';
  publicKey?: string;
  fingerprint?: string;
  os?: string;
  appVersion?: string;
  lastSeenAt?: string;
  createdAt: string;
  revokedAt?: string;
}

export interface CreateDeviceRequest {
  inviteCode: string;
  deviceName: string;
  os: string;
  appVersion: string;
  publicKey?: string;
  fingerprint?: string;
}

export interface CreateInviteRequest {
  ttlSeconds?: number;
  allowedRole?: string;
  scopes?: string[];
}

export interface InviteResponse {
  inviteId: string;
  inviteCode: string;
  expiresAt: string;
  qrPayload: string;
}

export interface HeartbeatRequest {
  deviceId: string;
  appVersion: string;
  collectorStatus: string;
  pendingUploadCount: number;
  watchedSourceCount: number;
}

File: apps/hub-service/src/models/device.rs
use serde::{Deserialize, Serialize};

#[allow(dead_code)]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Device {
    pub id: String,
    pub workspace_id: String,
    pub name: String,
    pub role: String,
    pub status: String,
    pub public_key: Option<String>,
    pub fingerprint: Option<String>,
    pub os: Option<String>,
    pub app_version: Option<String>,
    pub last_seen_at: Option<String>,
    pub created_at: String,
    pub revoked_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateDeviceRequest {
    pub invite_code: String,
    pub device_name: String,
    pub os: String,
    pub app_version: String,
    pub public_key: Option<String>,
    pub fingerprint: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateInviteRequest {
    pub ttl_seconds: Option<i64>,
    pub allowed_role: Option<String>,
    pub scopes: Option<Vec<String>>,
}

#[allow(dead_code)]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct InviteResponse {
    pub invite_id: String,
    pub invite_code: String,
    pub expires_at: String,
    pub qr_payload: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HeartbeatRequest {
    pub device_id: String,
    pub app_version: String,
    pub collector_status: String,
    pub pending_upload_count: i32,
    pub watched_source_count: i32,
}
